pennylane/measure.py

```diff
@@ -145,3 +145,42 @@ def var(op):
         QNode._current_context._append_op(op)
 
     return op
+
+
+def sample(op, n=None):
+    r"""Returns a sample of the supplied observable.
+
+    Args:
+        op (Observable): a quantum observable object
+        n (int): number of samples that should be obtained. Defaults to the
+            number of shots given as a parameter to the corresponding Device.
+    """
+    if not isinstance(op, Observable):
+        raise QuantumFunctionError(
+            "{} is not an observable: cannot be used with sample".format(op.name)
+        )
+
+    if QNode._current_context is not None:
+        # delete operations from QNode queue
+        QNode._current_context.queue.remove(op)
+
+    # set return type to be a sample
+    op.return_type = "sample"
+
+    # decorate the operation with the number of samples specified
+    if n is None:
+        if QNode._current_context is not None:
+            op.num_samples = QNode._current_context.device.shots
+    else:
+        op.num_samples = n
+
+    if n == 0:
+        raise ValueError("Calling sample with n = 0 is not possible.")
+    if n < 0 or not isinstance(n, int):
+        raise ValueError("The number of samples must be a positive integer.")
+
+    if QNode._current_context is not None:
+        # add observable to QNode observable queue
+        QNode._current_context._append_op(op)
```

Comment:
Looks good. This reminds me though that we should come up with a good solution to the operation queuing issue (see #161).